File: frontend/src/api/deviceConfig.ts
/** RTK Query endpoints for device configuration. */
import { api } from "./base";

export interface DeviceConfig {
  id: string;
  device_id: string;
  heartbeat_timeout_seconds: number;
  enabled: boolean;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface DeviceConfigUpdate {
  heartbeat_timeout_seconds?: number;
  enabled?: boolean;
  notes?: string;
}

export const deviceConfigApi = api.injectEndpoints({
  endpoints: (builder) => ({
    getDeviceConfig: builder.query<DeviceConfig, string>({
      query: (deviceId) => ({
        url: `/device-config/${deviceId}`,
        method: "GET",
      }),
      providesTags: (result, error, deviceId) => [{ type: "DeviceConfiguration", id: deviceId }],
    }),
    updateDeviceConfig: builder.mutation<DeviceConfig, { deviceId: string; data: DeviceConfigUpdate }>({
      query: ({ deviceId, data }) => ({
        url: `/device-config/${deviceId}`,
        method: "PATCH",
        body: data,
      }),
      invalidatesTags: (result, error, { deviceId }) => [
        { type: "DeviceConfiguration", id: deviceId },
        "Device",
      ],
    }),
  }),
});

export const { useGetDeviceConfigQuery, useUpdateDeviceConfigMutation } = deviceConfigApi;

File: backend/app/schemas/device_config.py
"""Device configuration schemas."""
from pydantic import BaseModel, Field
from uuid import UUID
from typing import Optional
from datetime import datetime


class DeviceConfigBase(BaseModel):
    heartbeat_timeout_seconds: int = Field(ge=30, le=86400, description="Timeout in seconds (30s to 24h)")
    enabled: bool = True
    notes: Optional[str] = None


class DeviceConfigCreate(DeviceConfigBase):
    device_id: UUID


class DeviceConfigUpdate(BaseModel):
    heartbeat_timeout_seconds: Optional[int] = Field(None, ge=30, le=86400)
    enabled: Optional[bool] = None
    notes: Optional[str] = None


class DeviceConfigResponse(DeviceConfigBase):
    id: UUID
    device_id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
